### DISCUSSION TITLE
A better way to define a function

### DISCUSSION THREAD

**Discussion by WithSJ** [CsqLang/Csq3#12] [Ideas]
I faced issue #10 
while testing it I think why we need Curley braces in this we need to define a function in a better way

so my idea is instead of using { } we should use the "ends" keyword

like this 

```
def myFunc() 
    print("Hello world")
ends myFunc
```

**Comment by AniketKumar2500** [comment]
Good idea so base/code_analysis.h now needs a upgrade!!

**Reply by WithSJ** [reply]
yes... we need to upgrade it

**Comment by AniketKumar2500** [comment]
Just give me some time upgrading it i got the logic!

**Reply by WithSJ** [reply]
good.. take your time

**Comment by AniketKumar2500** [comment]
I made a syntax for functions is it closed to english?
def fun() go
    ....
fun ends
what do you think?

**Reply by WithSJ** [reply]
Nice 👍

**Reply by WithSJ** [reply]
I think instead of "go" can we use "begin" 
That's because we start a code block whenever we want any kind of code block we can use it
And we can use it on class also

```
Class classname begin

classname ends
```

And if we want jump statement we can use it also

```
myblock begin
    print("Hello")
myblock ends

goto myblock 
``` 

**What you think about this.**

**Reply by AniketKumar2500** [reply]
Yeah it's a good idea.

**Reply by AniketKumar2500** [reply]
But it will become verbose to the user so lets don't apply it to Csq but i have changed go to starts.

**Reply by WithSJ** [reply]
don't worry I don't mind, that's just an Idea.

**Reply by AniketKumar2500** [reply]
Ok

**Comment by AniketKumar2500** [comment]
I have completed you could see it on repo.
